src/schema/v2/artist/shows.ts

```diff
@@ -34,92 +33,55 @@ export function showsWithDenyListedPartnersRemoved(shows) {
-  isReference: {
-    type: GraphQLBoolean,
-  },
-  size: {
-    type: GraphQLInt,
-    description: "The number of PartnerShows to return",
-  },
-  soloShow: {
-    type: GraphQLBoolean,
-  },
-  status: {
-    type: GraphQLString,
-  },
-  topTier: {
-    type: GraphQLBoolean,
-  },
-  visibleToPublic: {
-    type: GraphQLBoolean,
-  },
-  sort: {
-    type: ShowSorts,
-  },
-}
-
 export const ShowsConnectionField: GraphQLFieldConfig<
   { id: string },
   ResolverContext
 > = {
   type: ShowsConnection.connectionType,
-  args: pageable(ShowArgs),
-  resolve: ({ id }, args, { relatedShowsLoader }) => {
-    const gravityArgs = {
+  args: pageable({
+    active: { type: GraphQLBoolean },
+    atAFair: { type: GraphQLBoolean },
+    isReference: { type: GraphQLBoolean },
+    page: { type: GraphQLInt },
+    size: {
+      type: GraphQLInt,
+      description: "The number of PartnerShows to return",
+    },
+    soloShow: { type: GraphQLBoolean },
+    status: { type: GraphQLString },
+    topTier: { type: GraphQLBoolean },
+    visibleToPublic: { type: GraphQLBoolean },
+    sort: { type: ShowSorts },
+  }),
+  resolve: async ({ id }, args, { relatedShowsLoader }) => {
+    const { page, size, offset } = convertConnectionArgsToGravityArgs(args)
+
+    const { body, headers } = await relatedShowsLoader({
       active: args.active,
+      artist_id: id,
       at_a_fair: args.atAFair,
       is_reference: args.isReference,
-      size: args.size,
+      // /related endpoint doesn't support offset based navigation
+      page,
+      size,
       solo_show: args.soloShow,
+      sort: args.sort || "-end_at",
       status: args.status,
       top_tier: args.topTier,
+      total_count: true,
       visible_to_public: args.visibleToPublic,
-      sort: args.sort,
-    }
-    const pageOptions = convertConnectionArgsToGravityArgs(args)
-    const { page, size, offset } = pageOptions
-    return relatedShowsLoader(
-      defaults(
-        gravityArgs,
-        { page, size },
-        {
-          artist_id: id,
-          sort: "-end_at",
-          total_count: true,
-        }
-      )
-    )
-      .then(({ body, headers = {} }) => {
-        const allowListedShows = showsWithDenyListedPartnersRemoved(body)
-        return { body: allowListedShows, headers }
-      })
-      .then(({ body, headers }) => {
-        const totalCount = parseInt(headers["x-total-count"] || "0", 10)
-        const totalPages = Math.ceil(totalCount / size)
+    })
+
+    const totalCount = parseInt(headers["x-total-count"] || "0", 10)
+    const allowListedShows = showsWithDenyListedPartnersRemoved(body)
 
-        return merge(
-          {
-            pageCursors: createPageCursors(pageOptions, totalCount),
-            totalCount,
-          },
-          connectionFromArraySlice(body, args, {
-            arrayLength: totalCount,
-            sliceStart: offset,
-          }),
-          {
-            pageInfo: {
-              hasPreviousPage: page > 1,
```

Comment:
Oh interesting, it was manually computing this value. It's funny how this makes intuitive sense. I wonder why they didn't union type this thing.

Reply:
They = Facebook